packages/runner/lib/sdk/locks.ts

```diff
@@ -2,62 +2,336 @@
+const STATE_OFFSET = 0; // Int32: 0 = free, 1 = locked (4 bytes)
+const EXPIRY_OFFSET = 8; // BigInt64: Expiry timestamp (ms) (8 bytes) = STATE_OFFSET + STATE size + padding (4 bytes)
+const OWNER_LEN_OFFSET = 16; // Uint8: Length of owner string (1 byte) = EXPIRY_OFFSET + EXPIRY size
+const OWNER_STR_OFFSET = 17; // Uint8[255]: Owner string bytes (255 bytes) = OWNER_LEN_OFFSET + OWNER_LEN size
+const KEY_LEN_OFFSET = 17 + MAX_STRING_LEN; // Uint8: Length of key string (1 byte) = OWNER_STR_OFFSET + MAX_STRING_LEN
+const KEY_STR_OFFSET = 18 + MAX_STRING_LEN; // Uint8[255]: Key string bytes (255 bytes) = KEY_LEN_OFFSET + KEY_LEN size
+const BYTES_PER_LOCK = 18 + MAX_STRING_LEN * 2; // Total size of each lock slot (528 bytes)
+const BUFFER_SIZE = MAX_LOCKS * BYTES_PER_LOCK; // 256 * 528 = 135168 bytes
+const STATE_FREE = 0;
+const STATE_LOCKED = 1;
+
+const textEncoder = new TextEncoder();
+const textDecoder = new TextDecoder();
 
 export class Locks {
-    private store = new Map<string, Lock>();
+    private sab: SharedArrayBuffer;
+    // views for atomic operations
+    private int32View: Int32Array; // for state (free/locked)
+    private bigInt64View: BigInt64Array; // for expiry timestamp
+    private uint8View: Uint8Array; // for owner and key strings
 
-    public async tryAcquireLock({ owner, key, ttlMs }: { owner: string; key: string; ttlMs: number }): Promise<Result<boolean>> {
-        if (!owner || owner.length === 0 || owner.length > 255) {
-            return Err('Invalid lock owner (must be between 1 and 255 characters)');
+    private constructor(buffer: SharedArrayBuffer) {
+        if (buffer.byteLength !== BUFFER_SIZE) {
+            throw new Error(`Invalid buffer size. Expected ${BUFFER_SIZE}, got ${buffer.byteLength}`);
+        }
+        this.sab = buffer;
+        this.int32View = new Int32Array(this.sab);
+        this.bigInt64View = new BigInt64Array(this.sab);
+        this.uint8View = new Uint8Array(this.sab);
+    }
+    public static create(): Locks {
+        const buffer = new SharedArrayBuffer(BUFFER_SIZE);
+        const initialInt32View = new Int32Array(buffer);
+        const stateElementStride = BYTES_PER_LOCK / 4;
+        const stateElementOffset = STATE_OFFSET / 4;
+        for (let i = 0; i < MAX_LOCKS; i++) {
+            initialInt32View[i * stateElementStride + stateElementOffset] = STATE_FREE;
+        }
+        return new Locks(buffer);
+    }
+    public static fromBuffer(buffer: SharedArrayBuffer): Locks {
+        return new Locks(buffer);
+    }
+    public getBuffer(): SharedArrayBuffer {
+        return this.sab;
+    }
+
+    private getStateIndex(slotIndex: number): number {
+        return slotIndex * (BYTES_PER_LOCK / 4) + STATE_OFFSET / 4;
+    }
+    private getExpiryIndex(slotIndex: number): number {
+        return slotIndex * (BYTES_PER_LOCK / 8) + EXPIRY_OFFSET / 8;
+    }
+
+    private findSlotByKey(keyBytes: Uint8Array): number {
+        for (let i = 0; i < MAX_LOCKS; i++) {
```

Comment:
looking for a lock is O(n). 
Going for hashing and collision handling would be more complicated and given the limited size (256) I think this is a acceptable tradeoff. let me know if you disagree